esp/services/WsDeploy/WsDeployService.cpp

```diff
@@ -974,6 +974,17 @@ bool CWsDeployFileInfo::saveSetting(IEspContext &context, IEspSaveSettingRequest
       const char* pszOnChange = pSetting->queryProp("@onChange");
       const char* pszViewType = pSetting->queryProp("@viewType");
 
+      //
+      // Trim trailing blanks from ECL Agent queue name
+      if (!strcmp(pszAttrName, "wuQueueName") && strlen(pszNewValue))
+      {
+        char *pValue = const_cast<char *>(pszNewValue);
```

Comment:
This is ugly - it's a const char * because you don't own it, and casting it to char * to get around that is unsafe. You need to make a copy, modify the copy, and then call pSetting->setProp, or similar.

It is (quite rightly) undefined what the effect of modifying the string returned by queryProp is.